pkg/trustedresources/verify.go

```diff
@@ -75,20 +52,13 @@ func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.
 		ObjectMeta: tm,
 		Spec:       taskObj.TaskSpec(),
 	}
```

Comment:
why do we construct a new task object here? what if the original task has different values? im assuming this is being done to intentionally ignore some aspects of the Task when verifying the signature - this would be a good place for some comments to explain a bit more

Reply:
are there docs which explain what fields are ignored when this verification is performed? I didn't see anything in the trusted resources .md doc but we should explain those details there

Reply:
No users side data/values are ignored. But annotations added by k8s are ignored. I didn't add it into the doc b/c these annotations are not set by users. But I think it makes sense to add them.

Reply:
> why do we construct a new task object here? what if the original task has different values? im assuming this is being done to intentionally ignore some aspects of the Task when verifying the signature - this would be a good place for some comments to explain a bit more

remote resolution returns a task object, so we need to construct a `v1beta1.Task` here to objectmeta and spec from the task object. I may be wrong. We can discuss further

Reply:
discussed with @Yongxuanzhang  - action to revisit remote resolution and see if it could make sense to include the kind and api version in the values returned from remote resolution so they don't need to be reconstructed here (and maybe this logic could potentially work regardless of the api version in future? i.e. doesnt need to explicitly be aware of beta vs v1 types... 🤞 )